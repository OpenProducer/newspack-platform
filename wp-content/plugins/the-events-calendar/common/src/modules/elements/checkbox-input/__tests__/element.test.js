/**
 * External dependencies
 */
import React from 'react';
import { noop } from 'lodash';

/**
 * Internal dependencies
 */
import CheckboxInput from '../element.js';

describe( 'Input element', () => {
	it( 'Should render the component', () => {
		const component = renderer.create( <CheckboxInput /> );
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	it( 'Should render the component with checked', () => {
		const component1 = renderer.create( <CheckboxInput checked={ true } /> );
		expect( component1.toJSON() ).toMatchSnapshot();
		const component2 = renderer.create( <CheckboxInput checked={ false } /> );
		expect( component2.toJSON() ).toMatchSnapshot();
	} );

	it( 'Should render the component with class', () => {
		const component = renderer.create( <CheckboxInput className="checkbox-class" /> );
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	it( 'Should render the component with onChange handler', () => {
		const component = renderer.create( <CheckboxInput onChange={ noop } /> );
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	it( 'Should render the component with extra props', () => {
		const component = renderer.create( <CheckboxInput id="checkbox-id" /> );
		expect( component.toJSON() ).toMatchSnapshot();
	} );
} );
