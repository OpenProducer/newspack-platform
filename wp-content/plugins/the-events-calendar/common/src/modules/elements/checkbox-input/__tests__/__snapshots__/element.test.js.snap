// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input element Should render the component 1`] = `
<input
  className="tribe-editor__input tribe-editor__input--checkbox"
  type="checkbox"
/>
`;

exports[`Input element Should render the component with checked 1`] = `
<input
  checked={true}
  className="tribe-editor__input tribe-editor__input--checkbox"
  type="checkbox"
/>
`;

exports[`Input element Should render the component with checked 2`] = `
<input
  checked={false}
  className="tribe-editor__input tribe-editor__input--checkbox"
  type="checkbox"
/>
`;

exports[`Input element Should render the component with class 1`] = `
<input
  className="tribe-editor__input tribe-editor__input--checkbox checkbox-class"
  type="checkbox"
/>
`;

exports[`Input element Should render the component with extra props 1`] = `
<input
  className="tribe-editor__input tribe-editor__input--checkbox"
  id="checkbox-id"
  type="checkbox"
/>
`;

exports[`Input element Should render the component with onChange handler 1`] = `
<input
  className="tribe-editor__input tribe-editor__input--checkbox"
  onChange={[Function]}
  type="checkbox"
/>
`;
